extension-ui/background.js

```diff
@@ -193,3 +193,198 @@ function showNotification(site, num, options = {}) {
   
   chrome.notifications.create(notificationOptions);
 }
+
+let timeoutId = null;
+let notificationTimeoutId = null;
+let remainingTime = initialTime;
+let lastVisitTime = Date.now();
+let lastVisitDate = new Date().toDateString();
+let isSiteBlocked = false;
+let notificationShown = false;
+let currentTabId = null;
+let currentTabUrl = null;
+let isTabActive = false; 
+
+function updateStorage() {
+  chrome.storage.local.set({
+    lastVisitDate,
+    remainingTime,
+    isSiteBlocked,
+    notificationShown
+  });
+}
+
+function checkForNewDay() {
+  const currentDate = new Date().toDateString();
+  if (currentDate !== lastVisitDate) {
+    remainingTime = initialTime;
+    lastVisitDate = currentDate;
+    isSiteBlocked = false;
+    notificationShown = false;
+    updateStorage();
+  }
+}
+
+function startTimer() {
+  checkForNewDay();
+  clearTimeout(notificationTimeoutId);
+  clearTimeout(timeoutId);
+
+  const warningTime = remainingTime - notificationTime;
+  if (warningTime <= 0) {
```

Comment:
how can warning time be less than 0? when site's status is blocked user cannot browse in this site anymore
also when the user enters to blocked site he gets the oops page directly, you dont need to redirect here